controllers/result.go

```diff
@@ -22,10 +24,18 @@ func (c *ResultController) URLMapping() {
 
 func (c *ResultController) List() {
 	c.EnsureAuthenticatedUser()
-
 	c.TplName = "results/list.html"
-
 	web.ReadFromRequest(&c.Controller)
+
+	if c.CurrentUser != nil {
```

Comment:
Why we are checking the `CurrentUser` here? I think this checking should be done in `c.EnsureAuthenticatedUser()` so we don't need to check it after executing `c.EnsureAuthenticatedUser()`

Reply:
✅  removed in [beb1315](https://github.com/carryall/go-google-scraper-challenge/pull/65/commits/beb1315fa2bbede2da25dd42946489979fc7ad44)